cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go

```diff
@@ -676,3 +665,24 @@ func installedPackageAvailablePackageRef(rel *helmv2.HelmRelease) (*corev1.Avail
 		Context:    &corev1.Context{Namespace: repoNamespace},
 	}, nil
 }
+
+// ref https://fluxcd.io/docs/components/helm/helmreleases/
+func helmReleaseName(key types.NamespacedName, rel *helmv2.HelmRelease) types.NamespacedName {
+	helmReleaseName := rel.Spec.ReleaseName
+	// according to docs ReleaseName is optional and defaults to a composition of
+	// '[TargetNamespace-]Name'.
+	if helmReleaseName == "" {
+		targetNamespace := rel.Spec.TargetNamespace
+		// according to docs targetNamespace is optional and defaults to the namespace of the HelmRelease
```

Comment:
Just noting, not necessary for this fix, but for the related #4189, that the behaviour flux shows is slightly different to below (see my notes from updating the flux CR on #4172, or your notes on #4189) but still consistent with the comment above: the namespace targeted by the `HelmRelease` defaults to the namespace of the `HelmRelease`, but flux doesn't set the `targetNamespace` field with that value, it remains empty as specified. And when the `targetNamespace` is empty/unset, the helm release name is just the `HelmRelease.metadata.name` (ie. not prefixed by the namespace at all, and so different to below). This is why I was unsure why we were explicitly setting the `targetNamespace` given flux's default behaviour.

Reply:
> Just noting, not necessary for this fix, but for the related #4189, that the behaviour flux shows is slightly different to below (see my notes from updating the flux CR on #4172, or your notes on #4189) but still consistent with the comment above: the namespace targeted by the `HelmRelease` defaults to the namespace of the `HelmRelease`, but flux doesn't set the `targetNamespace` field with that value, it remains empty as specified. And when the `targetNamespace` is empty/unset, the helm release name is just the `HelmRelease.metadata.name` (ie. not prefixed by the namespace at all, and so different to below). This is why I was unsure why we were explicitly setting the `targetNamespace` given flux's default behaviour.

noted